### DISCUSSION TITLE
What if the retrieved ddl is not related to the user question?

### DISCUSSION THREAD

**Discussion by Jackie-shi** [vanna-ai/vanna#450] [Q&A]
When do 'ask', the process here is to use vector database to get the most related DDL. But there is a situation that all of the ddl has nothing to do with the user question, which means that the final SQL Vana generates will definitely WRONG!

So I wonder if there is any mechanism or post-process to check or evaluate the retrieved ddl.🤔

THX!

**Comment by Jackie-shi** [comment]
What's more, when you do embedding, the raw ddl is not a natural language so the problem between ddl and user question is that they cannot be aligned in semantic space. It may make the result inaccurate!

**Comment by zainhoda** [comment]
Extra information is not usually detrimental to the SQL output, particularly if you have a lot of question-SQL pairs

**Comment by koalababy2024** [comment]
I have the same concern - there is almost no semantic similarity between user questions and DDL statements. How can we directly use the user's question to search for similar DDL statements through vector similarity?

**Reply by MichaelMMeskhi** [reply]
You can try schema linking. Basically link the relevant schemas to their related SQL/Question pairs. So whenever a SQL/Question pair is retrieved as relevant to the user question, their relevant schemas will also tag along.
